torch/autograd/__init__.py

```diff
@@ -140,6 +140,10 @@ def grad(outputs, inputs, grad_outputs=None, retain_graph=None, create_graph=Fal
         inputs)
 
 
+def is_checkpoint_valid():
```

Comment:
We had a really long, detailed discussion about why this was necessary, and the source code makes no mention of it at all. It would be good to copy paste relevant snippets from the discussion into a Note in the source code, so anyone who sees `is_checkpoint_valid` can know the motivation behind it. Even better to have references to the Note for all sites mentioning checkpoint_valid.

Reply:
sounds good, will do so.